core/arch/arm/kernel/secure_partition.c

```diff
@@ -0,0 +1,751 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Arm Limited. All rights reserved.
+ * Copyright (c) 2019, Linaro Limited
+ */
+
+#include <kernel/abort.h>
+#include <kernel/secure_partition.h>
+#include <kernel/user_mode_ctx.h>
+#include <mm/fobj.h>
+#include <mm/mobj.h>
+#include <mm/tee_mmu.h>
+#include <pta_stmm.h>
+#include <tee/tee_svc.h>
+#include <tee/tee_pobj.h>
+#include <zlib.h>
+
+#include <ffa.h>
+#include "thread_private.h"
+#include <tee/tee_svc_storage.h>
+#include <crypto/crypto.h>
+#include <tee_api_defines_extensions.h>
+
+static const TEE_UUID stmm_uuid = PTA_STMM_UUID;
+
+static const uint16_t stmm_id = 1U;
+static const uint16_t stmm_pta_id = 2U;
+static const uint16_t mem_mgr_id = 3U;
```

Comment:
I don't see a way for StMM to query these numbers so I expect that they are hard coded in StMM too. Perhaps add a comment that this is part of the ABI towards StMM?

Reply:
This isn't an ABI exactly. Those are indeed hardcoded in StMM at the moment, but isn't the final idea to make them discoverable once we get the full FFA spec in place?

Reply:
It's my view also that this will become discoverable at some point, until then it is an ABI though. If we change these numbers it will break old StMM binaries.

Reply:
Ok, I'll add a comment describing this